src/platform/input/game-pads.js

```diff
@@ -68,62 +166,754 @@ const MAPS = {
+                        return actuator.playEffect(actuator.type, {
+                            duration,
+                            startDelay,
+                            strongMagnitude,
+                            weakMagnitude
+                        });
+                    } else if (actuator.pulse) {
+                        await sleep(startDelay);
+                        return actuator.pulse(intensity, duration);
+                    }
+
+                    return false;
+                })
+            );
+
+            return results.some(r => r === true || r === 'complete');
+        }
+
+        return false;
+    }
+
+    /**
+     * Retrieve a button from its index name.
+     *
+     * @param {string} indexName - The index name to return the button for.
+     * @returns {GamePadButton|null} The button for the searched index. Can be null.
+     * @ignore
+     */
+    _getButton(indexName) {
+        const synthesizedButton = this.map.synthesizedButtons[indexName];
+        if (synthesizedButton) {
+            const { axis, max, min } = synthesizedButton;
+
+            return new GamePadButton(
+                Math.abs(math.clamp(this._axes[axis] ?? 0, min, max)),
+                Math.abs(math.clamp(this._previousAxes[axis] ?? 0, min, max))
+            );
+        }
+
+        const button = this._buttons[MAPS_INDEXES.buttons[indexName]];
+
+        if (button) {
+            return button;
+        }
+
+
+        return null;
+    }
+
+    /**
+     * Retrieve a button from its index.
+     *
+     * @param {number} index - The index to return the button for.
+     * @returns {GamePadButton|null} The button for the searched index. Can be null.
+     */
+    getButton(index) {
+        const buttons = this.map.buttons;
+        return buttons && buttons[index] ? this._getButton(buttons[index]) : null;
+    }
+
+    /**
+     * Returns true if the button is pressed.
+     *
+     * @param {number} button - The button to test, use constants {@link PAD_FACE_1}, etc.
+     * @returns {boolean} True if the button is pressed.
+     */
+    isPressed(button) {
+        const b = this.getButton(button);
+        return b && b.pressed;
+    }
+
+    /**
+     * Return true if the button was pressed since the last update.
+     *
+     * @param {number} button - The button to test, use constants {@link PAD_FACE_1}, etc.
+     * @returns {boolean} Return true if the button was pressed, false if not.
+     */
+    wasPressed(button) {
+        const b = this.getButton(button);
+        return b && b.wasPressed;
+    }
+
+    /**
+     * Return true if the button was released since the last update.
+     *
+     * @param {number} button - The button to test, use constants {@link PAD_FACE_1}, etc.
+     * @returns {boolean} Return true if the button was released, false if not.
+     */
+    wasReleased(button) {
+        const b = this.getButton(button);
+        return b && b.wasReleased;
```

Comment:
```suggestion
        return this.getButton(button)?.wasReleased || false;
```

Just to make it type safe, right now this is possibly returning `null`:

![image](https://user-images.githubusercontent.com/5236548/220654609-b1801942-586e-497e-98dc-e07c38a70e4e.png)

Same for isPressed/wasPressed/isTouched/wasTouched

Reply:
Oh that's interesting, I was thinking that a logical operator was always returning boolean. That's fixed, thanks a lot!